scripts/lintcommit.lua

```diff
@@ -0,0 +1,34 @@
+local current_branch = vim.trim(vim.fn.system({'git', 'branch', '--show-current'}))
+local ancestor_commit = vim.trim(vim.fn.system({'git', 'merge-base', 'master', current_branch}))
+local commits_on_current_pr = vim.trim(vim.fn.system({'git', 'rev-list', ancestor_commit .. ".." .. current_branch}))
+
+local all_commits = {}
+for substring in commits_on_current_pr:gmatch("%S+") do
+   table.insert(all_commits, substring)
```

Comment:
Looks like `commits_on_current_pr` and `all_commits` are semantically the same but just different in their structure. This is an example of the peril of "premature explanation": `commits_on_current_pr` has an elaborate (prematurely-explained) name and but is actually just an intermediate container. Then `all_commits` struggles to find a different name.

I would rename them as:
- `commits_on_current_pr` => `commits_str`
- `all_commits` => `commits`

(Of course, this script has a small scope so it "doesn't matter", but I mention this for reference  in future/bigger contributions.)

Reply:
Good catch. I'll keep it in mind, thank you.